### PR TITLE
EZP-24824: As an editor, I want to be able to see and browse the subitems of a Location.

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-24824

# Description

This patch provides the very first sub-items list. It performs quite well (one HTTP request to build it) thanks to the Location search exposed *via* the REST views. The bug issue right now is that [the list is not sorted](https://jira.ez.no/browse/EZP-24849) because [the REST views do not support that yet](https://jira.ez.no/browse/EZP-24315) (see also https://github.com/ezsystems/ezpublish-kernel/pull/1417).

Screencast: https://www.youtube.com/watch?v=0alRMQZTcmg

## Tasks

* [x] Fix the bug in Location where the id is always undefined
* [x] Add a location search plugin to be able to trigger a Location search
* [x] Implement the asynchronous view to list the sub items
* [x] Handle the case where the content is not a container
* [x] Pagination

# Tests

manual + unit test

### CODE REVIEW

**Comment by dpobel** [comment]
ping @mhyndle @StephaneDiot @yannickroger @andrerom

**Comment by andrerom** [comment]
+1 but @inakijv needs to start to think about some design tweaks I think :)

**Comment by dpobel** [comment]
in an ideal world, this should come with the specifications, in an ideal world...

**Comment by andrerom** [comment]
:)

File: Resources/public/css/views/subitemlist.css [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+.ez-view-subitemlistview .ez-subitemlist-content {
+    margin: 1em;
+}
+
+.ez-view-subitemlistview .ez-subitemlist-table {
+    width: 100%;
+}
+
+.ez-view-subitemlistview .ez-subitemlist-navigation {
+    display: flex;
```
Comment (dpobel) [inline Resources/public/css/views/subitemlist.css]: self note: missing browser prefixes

File: Resources/public/js/views/ez-subitemlistview.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+         * @method _hasPages
+         * @private
+         * @return {Boolean}
+         */
+        _hasPages: function () {
+            return this.get('location').get('childCount') > this.get('limit');
+        },
+
+        /**
+         * Checks whether the user is on the last "page".
```
Comment (mhyndle) [inline Resources/public/js/views/ez-subitemlistview.js]: `is on first "page".` and the same in `@method` doc below

**Comment by mhyndle** [comment]
+1, besides inline comment above

File: Resources/public/js/views/ez-subitemlistview.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+     * Provides the subitem list view.
+     *
+     * @module ez-subitemlistview
+     */
+    Y.namespace('eZ');
+
+    var IS_PAGE_LOADING = 'is-page-loading',
+        IS_DISABLED = 'is-disabled';
+
+    function linkIsDisabled(link) {
```
Comment (yannickroger) [inline Resources/public/js/views/ez-subitemlistview.js]: Should this need JSdoc ?

**Comment by yannickroger** [comment]
+1

File: Resources/public/js/views/ez-subitemlistview.js (hunk 3 above)
Comment (dpobel) [inline Resources/public/js/views/ez-subitemlistview.js]: it's not exposed in any way (really private) and I guess the name is clear enough

**Pull request merged**